API/Controllers/TournamentsController.Admin.cs

```diff
@@ -0,0 +1,162 @@
+using API.DTOs;
+using API.Utilities;
+using API.Utilities.Extensions;
+using Database.Entities;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Controllers;
+
+public partial class TournamentsController
+{
+    /// <summary>
+    /// Creates an admin note for a tournament
+    /// </summary>
+    /// <param name="id">Tournament id</param>
+    /// <response code="401">If the requester is not properly authorized</response>
+    /// <response code="404">If a tournament matching the given id does not exist</response>
+    /// <response code="400">If the authorized user does not exist</response>
+    /// <response code="200">Returns the created admin note</response>
+    [HttpPost("{id:int}/notes")]
+    [Authorize(Roles = $"{OtrClaims.Admin}")]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType<AdminNoteDTO>(StatusCodes.Status200OK)]
+    public async Task<IActionResult> CreateAdminNoteAsync(int id, [FromBody] string note)
```

Comment:
In testing, I hard-coded my user id (20). The `adminUsername` field returned an empty string. From what I can see in the current code, there isn't a way for the `CreateAsync()` method to link back to a user. 
![CleanShot 2024-09-28 at 11 48 20@2x](https://github.com/user-attachments/assets/b85b074d-fd56-4c2d-9aff-c8909f2ea84b)

Reply:
I see, I guess I was expecting the returned entity to have a populated user object? I imagine the following:
- Admin creates note
- Note is displayed on entity (immediately, no additional GET call required).

I think it makes sense to have a populated `AdminUser` here then.

Reply:
We can load the navigation by force after the entity is created or we can simply save it to the database, and then immediately fetch it from the database using `AdminNoteRepository.GetAsync<T>(createdNote.Id)` since that method includes the required navigations. I think the later is probably best.

Reply:
Yeah I agree with you. So, basically just adding a call above the `return mapper.Map` which calls `.GetAsync` on the newly created entity to load the user navigation. If so that sounds good.